jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/UrlEncoded.java

```diff
@@ -358,12 +358,12 @@ public static boolean decodeUtf8To(String query, int offset, int length, BiConsu
      * @param offset the offset at which query parameters start.
      * @param length the length of query parameters string to parse.
      * @param adder the method to call to add decoded parameters.
-     * @param allowTruncatedUtf8 if {@code true} allow truncated UTF-8 and insert the replacement character.
+     * @param allowBadPercent if {@code true} allow bad percent encoding.
      * @param allowBadUtf8 if {@code true} allow bad UTF-8 and insert the replacement character.
      * @return {@code true} if the string was decoded without any bad UTF-8
      * @throws org.eclipse.jetty.util.Utf8StringBuilder.Utf8IllegalArgumentException if there is illegal UTF-8 and `allowsBadUtf8` is {@code false}
      */
-    public static boolean decodeUtf8To(String query, int offset, int length, BiConsumer<String, String> adder, boolean allowTruncatedUtf8, boolean allowBadUtf8)
+    public static boolean decodeUtf8To(String query, int offset, int length, BiConsumer<String, String> adder, boolean allowBadPercent, boolean allowBadUtf8, boolean allowTruncatedUtf8)
```

Comment:
If there is the possibility of more booleans, consider moving to an `EnumSet`.

Reply:
what... you mean like a compliance mode is :) 
It would probably be better to move these methods to jetty-http and just use the compliance mode directly.